introduction_to_amazon_algorithms/jumpstart-foundation-models/text-generation-benchmarking/benchmarking/runner.py

```diff
@@ -93,29 +93,60 @@ def _run_benchmarking_tests(
         metrics_throughput: Dict[str, Any] = {}
         metrics_concurrency: Dict[str, Any] = {}
 
-        endpoint_description = self._sagemaker_client.describe_endpoint(predictor.endpoint_name)
-        endpoint_config_name = endpoint_description["EndpointConfigName"]
-        endpoint_config_description = self._sagemaker_client.describe_endpoint_config(endpoint_config_name)
-        model_description = self._sagemaker_client.describe_model(predictor.endpoint_name)
-
-        production_variant = endpoint_config_description["ProductionVariants"][0]
-        primary_container = model_description.get("PrimaryContainer")
-
-        instance_type = production_variant["InstanceType"]
-        price_per_instance = self._pricing_client.get_price_per_unit(instance_type, SM_SESSION._region_name)
-        price_per_endpoint = production_variant["InitialInstanceCount"] * price_per_instance
-        metrics_pricing = {
-            "PricePerInstance": price_per_instance,
-            "PricePerEndpoint": price_per_endpoint,
-        }
-
-        creation_time: datetime = endpoint_description["CreationTime"]
-        last_modified_time: datetime = endpoint_description["LastModifiedTime"]
-        metrics_time = {
-            "CreationTime": creation_time.isoformat(),
-            "LastModifiedTime": last_modified_time.isoformat(),
-            "DeploymentTime": (last_modified_time - creation_time).seconds,
-        }
+        if hasattr(predictor, "predictor") and predictor.predictor is not None:
```

Comment:
nit: is `hasattr(predictor, "predictor")` necessary here since it has a `None` default value in the constructor?